Fixed no power call back messages and unable to control the buzzer

634296207 opened the pull request:
Troubleshooting No power call back information

634296207 committed:
Update target.h

Troubleshooting No power call back information

634296207 commented:
It has been tested with motor RPM, but no buzzer control

lida2003 commented:
Hi @634296207 

What do you really wanna do with this PR?

title: " Update target.h #7 "
description: "Troubleshooting No power call back information"
first comment: "It has been tested with motor RPM, but no buzzer control"

Can you provie a more specific tile and clarify the description?

634296207 committed:
Update config.c

1. Resolve the problem that the buzzer does not sound
2. Serial port 4 Add the image upload

634296207 committed:
Update target.h

Modify the TX5 and RX5 pins

634296207 commented:
1. Resolve the problem that the buzzer does not sound
2. Serial port 4 Add the VTX
3. Modify the TX5 and RX5 pins

lida2003 commented:
> 1. Resolve the problem that the buzzer does not sound
> 
>     2. Serial port 4 Add the VTX
> 
>     3. Modify the TX5 and RX5 pins

How about self test goes?

634296207 committed:
Update target.h

Add galvanometer ratio 206

634296207 commented:
PR self-test passed

634296207 commented:
INAV7.0.0 F405V3 SD V3 self test passed.

lida2003 reviewed (changes requested):
Why PD02 is Tx？

634296207 committed:
Update target.h

TX5 is PC12 and RX5 is PD2

634296207 commented:
TX5 is PC12 and RX5 is PD2

lida2003 commented on src/main/target/AOCODARCF4V3/config.c:
```diff
@@ -1,4 +1,3 @@
-/*
```
Not Good! Is it able to compile?

634296207 replied:
Verified, ready to compile

lida2003 replied:
OK. Now we have a bug.

```
daniel@palfocus:~/Aocoda-RC/inav/build$ cmake ..
-- found arm-none-eabi-gcc 10.3.1 at /home/daniel/Aocoda-RC/inav/tools/gcc-arm-none-eabi-10.3-2021.10/bin/arm-none-eabi-gcc
-- toolchain: arm-none-eabi, WARNINGS_AS_ERRORS: OFF
-- Could not find openocd, debugging won't be available
-- DEBUG_HARDFAULTS: OFF, SEMIHOSTING: OFF
-- DEBUG_HARDFAULTS: OFF, SEMIHOSTING: OFF
-- 191 targets (150 for release) found for toolchain arm-none-eabi
-- Build type: RelWithDebInfo
-- Configuring done
-- Generating done
-- Build files have been written to: /home/daniel/Aocoda-RC/inav/build
daniel@palfocus:~/Aocoda-RC/inav/build$ make AOCODARCF4V3_SD
Generating AOCODARCF4V3_SD/settings_generated.h, AOCODARCF4V3_SD/settings_generated.c
Scanning dependencies of target AOCODARCF4V3_SD.elf
Consolidate compiler generated dependencies of target AOCODARCF4V3_SD.elf
Building ASM object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/startup/startup_stm32f40xx.s.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/drivers/bus_spi.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/drivers/serial_uart.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/system_stm32f4xx.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/config/config_streamer_stm32f4.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/config/config_streamer_ram.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/config/config_streamer_extflash.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/drivers/adc_stm32f4xx.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/drivers/bus_i2c_stm32f40x.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/drivers/serial_uart_stm32f4xx.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/drivers/system_stm32f4xx.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/drivers/timer.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/drivers/timer_impl_stdperiph.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/drivers/timer_stm32f4xx.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/drivers/uart_inverter.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/drivers/dma_stm32f4xx.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/drivers/sdcard/sdmmc_sdio_f4xx.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/misc.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_adc.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_dac.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_dcmi.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_dfsdm.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_dma.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_dma2d.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_exti.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_flash.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_fsmc.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_gpio.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_i2c.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_iwdg.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_ltdc.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_pwr.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_rcc.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_rng.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_rtc.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_sdio.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_spi.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_syscfg.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_tim.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_usart.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/__/__/__/__/lib/main/STM32F4/Drivers/STM32F4xx_StdPeriph_Driver/src/stm32f4xx_wwdg.c.obj
Building C object src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/config.c.obj
/home/daniel/Aocoda-RC/inav/src/main/target/AOCODARCF4V3/config.c:1:9: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'file'
    1 |  * This file is part of Cleanflight.
      |         ^~~~
/home/daniel/Aocoda-RC/inav/src/main/target/AOCODARCF4V3/config.c:1:9: error: unknown type name 'file'
/home/daniel/Aocoda-RC/inav/src/main/target/AOCODARCF4V3/config.c:9:30: error: unknown type name 'without'
    9 |  * but WITHOUT ANY WARRANTY; without even the implied warranty of
      |                              ^~~~~~~
/home/daniel/Aocoda-RC/inav/src/main/target/AOCODARCF4V3/config.c:9:43: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'the'
    9 |  * but WITHOUT ANY WARRANTY; without even the implied warranty of
      |                                           ^~~
/home/daniel/Aocoda-RC/inav/src/main/target/AOCODARCF4V3/config.c:9:43: error: unknown type name 'the'
In file included from /home/daniel/Aocoda-RC/inav/tools/gcc-arm-none-eabi-10.3-2021.10/arm-none-eabi/include/stdint.h:14,
                 from /home/daniel/Aocoda-RC/inav/tools/gcc-arm-none-eabi-10.3-2021.10/lib/gcc/arm-none-eabi/10.3.1/include/stdint.h:9,
                 from /home/daniel/Aocoda-RC/inav/src/main/target/AOCODARCF4V3/config.c:16:
/home/daniel/Aocoda-RC/inav/tools/gcc-arm-none-eabi-10.3-2021.10/arm-none-eabi/include/sys/_stdint.h:20:9: error: unknown type name '__int8_t'
   20 | typedef __int8_t int8_t ;
      |         ^~~~~~~~
make[3]: *** [src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/build.make:657: src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/config.c.obj] Error 1
make[2]: *** [CMakeFiles/Makefile2:12409: src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.elf.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:12442: src/main/target/AOCODARCF4V3/CMakeFiles/AOCODARCF4V3_SD.dir/rule] Error 2
make: *** [Makefile:2503: AOCODARCF4V3_SD] Error 2
```

lida2003 commented:
> TX5 is PC12 and RX5 is PD2

Did you verify issues above in <Aocoda-RC-F4V3-V3HW-inav_sd_cd34460-V2.pdf>again, when you modify the PD02?

634296207 commented:
> > TX5 is PC12 and RX5 is PD2
> 
> Did you verify issues above in <Aocoda-RC-F4V3-V3HW-inav_sd_cd34460-V2.pdf>again, when you modify the PD02?
> 
> Does the above commit can be compiled?

Verified, ready to compile
